### PR TITLE
feat: Chat UI - Messaging Cards, Bottom Navigation, Conversation List, Sub Modals

### PR DESCRIPTION
**Pull request by ice-kreios**
### What does this PR do?
This PR implements the foundational UI components for the Chat feature

### Changes brought by this PR
* Messaging Cards for All Types
* Conversation List UI
* Chat Bottom Bar in different states
* Sub Chat Modals
* Voice Listening and Recording
* Enabling VisualDetector to pause/mute content
* Chat Blank/Empty Views

https://github.com/user-attachments/assets/c44bac50-3a24-42b6-88ee-551d05d34255

### CODE REVIEW

File: assets/svg/icon_chat_back.svg
Comment (ice-kreios) [inline assets/svg/icon_chat_back.svg:1]: waiting response from designer to be sure which one is valid.

File: lib/app/features/chat/models/conversation_type.dart
[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/features/chat/models/conversation_type.dart:1]: fixed

[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/features/chat/models/conversation_type.dart:1]: ✅

File: lib/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            isMe ? context.theme.appColors.onPrimaryAccent : context.theme.appColors.primaryText,
+        fixedWaveColor: context.theme.appColors.sheetLine,
+        seekLineColor: Colors.transparent,
+        waveThickness: 1.0.s,
+        spacing: 2.0.s,
+      ),
+      [isMe],
+    );
+
+    useEffect(
```
[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart:53]: Thanks for the suggestion! The state is only used internally here, so a provider might be unnecessary.

File: lib/app/features/chat/messages/views/components/audio_loading_indicator/audio_loading_indicator.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.all(1.0.s),
+      child: SizedBox.square(
+        dimension: 18.0.s,
+        child: CircularProgressIndicator(
```
[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/features/chat/messages/views/components/audio_loading_indicator/audio_loading_indicator.dart:17]: Noticed this differs from the Figma design, but to keep the loading widget consistent, I’ll check with the designer for confirmation.

File: lib/app/features/chat/messages/views/components/message_types/audio_message.dart [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+          isMe ? context.theme.appColors.onPrimaryAccent : context.theme.appColors.primaryText,
+      fixedWaveColor: context.theme.appColors.sheetLine,
+      seekLineColor: Colors.transparent,
+      waveThickness: 1.0.s,
+      spacing: 2.0.s,
+    );
+
+    useEffect(
+      () {
+        getApplicationDocumentsDirectory().then((value) {
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/audio_message.dart:44]: I would propose moving this to `MediaService`

File: lib/app/features/chat/messages/views/components/message_types/audio_message.dart [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      isMe: isMe,
+      contentPadding: EdgeInsets.all(12.0.s),
+      child: VisibilityDetector(
+        key: ValueKey(audioUrl),
+        onVisibilityChanged: (info) {
+          if (info.visibleFraction == 0) {
+            playerController.value.pausePlayer();
+          }
+        },
+        child: Row(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/audio_message.dart:83]: Let's create a standalone widget for Row

File: lib/app/features/chat/messages/views/components/message_types/audio_message.dart [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+          crossAxisAlignment: CrossAxisAlignment.end,
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            GestureDetector(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/audio_message.dart:87]: create please a class for this button

File: lib/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            ),
+            MessageTimeStamp(isMe: isMe),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+Future<void> _initializePlayer(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart:108]: let's create a `AudioService` and move logic related to the audio to that service

File: lib/app/features/chat/messages/views/components/message_types/money_receive_message.dart [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return MessageItemWrapper(
+      contentPadding: EdgeInsets.symmetric(
+        horizontal: 12.0.s,
+        vertical: 12.0.s,
+      ),
+      isMe: isMe,
+      child: Column(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/money_receive_message.dart:22]: Could you please split this widget tree into smaller classes for better readability?

File: lib/app/features/chat/messages/views/components/message_types/money_request_message.dart [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return MessageItemWrapper(
+      contentPadding: EdgeInsets.symmetric(
+        horizontal: 12.0.s,
+        vertical: 12.0.s,
+      ),
+      isMe: isMe,
+      child: Column(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/money_request_message.dart:22]: the same, split please on the smaller classes

File: lib/app/features/chat/messages/views/components/message_types/poll_message/poll_result_message.dart [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+              mockPoll.question,
+              style: context.theme.appTextThemes.body2.copyWith(
+                color: isMe
+                    ? context.theme.appColors.onPrimaryAccent
+                    : context.theme.appColors.primaryText,
+              ),
+            ),
+          ),
+          SizedBox(height: 10.0.s),
+          ListView.separated(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/poll_message/poll_result_message.dart:31]: extract to the standalone widgets

File: lib/app/features/chat/messages/views/components/message_types/profile_share_message/profile_share_message.dart [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return MessageItemWrapper(
+      contentPadding: EdgeInsets.symmetric(
+        horizontal: 12.0.s,
+        vertical: 12.0.s,
+      ),
+      isMe: isMe,
+      child: Row(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/profile_share_message/profile_share_message.dart:23]: split into the smaller widgets

File: lib/app/features/chat/messages/views/components/message_types/url_message.dart [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+              Text(
+                url,
+                style: context.theme.appTextThemes.body2.copyWith(
+                  color: isMe
+                      ? context.theme.appColors.onPrimaryAccent
+                      : context.theme.appColors.primaryAccent,
+                ),
+              ),
+              if (isMetaAvailable())
+                IntrinsicHeight(
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/message_types/url_message.dart:57]: refactor please this widget tree into the smaller classes

File: lib/app/features/chat/messages/views/components/messaging_bottom_bar/messaging_bottom_bar.dart [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/chat/messages/views/components/messaging_bottom_bar/messaging_bottom_bar_initial.dart';
+import 'package:ion/app/features/chat/messages/views/components/messaging_bottom_bar/messaging_bottom_bar_recording.dart';
+import 'package:ion/app/features/chat/model/messaging_bottom_bar_state.dart';
+import 'package:ion/app/features/chat/providers/messaging_bottom_bar_state_provider.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+part './components/action_button.dart';
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/components/messaging_bottom_bar/messaging_bottom_bar.dart:14]: Maybe create a `barrel file`?

File: lib/app/features/chat/messages/views/pages/messages_page.dart [hunk 13]
```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/chat/messages/views/components/chat_date_header_text/chat_date_header_text.dart';
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/pages/messages_page.dart:6]: create please a barrel file for these components to simplify imports

File: lib/app/features/chat/messages/views/pages/messages_page.dart [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+          child: Column(
+            children: [
+              const MessagingHeader(),
+              // const MessagingEmptyView(),
+              Expanded(
+                child: ColoredBox(
+                  color: context.theme.appColors.primaryBackground,
+                  child: ListView.separated(
+                    itemBuilder: (context, index) {
+                      switch (index) {
```
Comment (ice-alcides) [inline lib/app/features/chat/messages/views/pages/messages_page.dart:46]: Can you refactor this switch condition?

Comment (ice-kreios) [reply lib/app/features/chat/messages/views/pages/messages_page.dart:46]: it is just for mocking/testing, when we bind the backend, ll be refactored totally. is it ok for you @ice-alcides ?

Comment (ice-alcides) [reply lib/app/features/chat/messages/views/pages/messages_page.dart:35]: sure, understood

**Review by ice-alcides — APPROVED** [review]

File: lib/app/components/overlay_menu/overlay_menu.dart [hunk 15]
```diff
@@ -30,20 +31,32 @@ class OverlayMenu extends HookWidget {
 
         return GestureDetector(
           behavior: HitTestBehavior.translucent,
-          onTap: overlayPortalController.hide,
+          onTap: () {
+            animationController.reverse().whenComplete(overlayPortalController.hide);
```
Comment (ice-tychon) [inline lib/app/components/overlay_menu/overlay_menu.dart:35]: Consider creating functions to show and hide the menu.

File: lib/app/features/chat/messages/views/components/mesage_timestamp/mesage_timestamp.dart [hunk 16]
```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+class MessageTimeStamp extends StatelessWidget {
```
Comment (ice-tychon) [inline lib/app/features/chat/messages/views/components/mesage_timestamp/mesage_timestamp.dart:7]: This widget contains not only the timestamp but also `iconMessageRead`, which may be confusing. Please split it into two different widgets or rename this one.

File: lib/app/features/chat/messages/views/components/message_types/poll_message/poll_result_message.dart [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+  Widget build(BuildContext context) {
+    return ListView.separated(
+      shrinkWrap: true,
+      physics: const NeverScrollableScrollPhysics(),
+      itemBuilder: (context, index) {
+        final option = mockPoll.options[index];
+
+        final percentage = option.votes /
+            mockPoll.options.map((e) => e.votes).reduce((value, element) => value + element);
+        return Stack(
```
Comment (ice-tychon) [inline lib/app/features/chat/messages/views/components/message_types/poll_message/poll_result_message.dart:69]: Extract to a separate widget

File: lib/app/features/chat/messages/views/components/messaging_bottom_bar/components/more_content_view.dart [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+      height: moreContentHeight,
+      width: double.infinity,
+      color: context.theme.appColors.secondaryBackground,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: [
+              Column(
```
Comment (ice-tychon) [inline lib/app/features/chat/messages/views/components/messaging_bottom_bar/components/more_content_view.dart:23]: Extract each `Column` to a separate widget to improve readability.

Comment (ice-tychon) [reply lib/app/features/chat/messages/views/components/messaging_bottom_bar/components/more_content_view.dart:23]: Or, as I can see, they are similiar, so we need only 1 widget with  Icon and Label params.

File: lib/app/features/chat/views/pages/new_chat_modal/new_chat_modal.dart [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+                      child: SearchInput(
+                        textInputAction: TextInputAction.search,
+                      ),
+                    ),
+                  ],
+                ),
+                SizedBox(height: 12.0.s),
+                Row(
+                  children: [
+                    Expanded(
```
Comment (ice-tychon) [inline lib/app/features/chat/views/pages/new_chat_modal/new_chat_modal.dart:45]: These Expanded widgets look similiar, only Icon, OnPressed and Label are different. Consider extracting to a separate widget.

**Review by ice-tychon — APPROVED** [review]

**Review by ice-tychon — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Comment by Zaid-maker** [comment]
Is this coming to the ICE app?

**Pull request merged**